modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/edit_kaleo_definition_version.jsp

```diff
@@ -711,44 +661,6 @@ String successMessageKey = KaleoDesignerPortletKeys.KALEO_DESIGNER + "requestPro
 											}
 										</c:when>
 									</c:choose>
-
```

Comment:
Actually.... where did all this code go, @julien, @markocikos?

Reply:
... it looks like I deleted it when solving a conflict.
Let me put that back how it should

Reply:
This block of code was suppose to refresh portlet on modal close:
```
openerWindow.Liferay.Portlet.refresh(
	'#p_p_id<%= HtmlUtil.escapeJS(portletResourceNamespace) %>'
);
```
There is no refresh in master, I am not sure why, looks like `portletAjaxable` patters was not configured correctly. [This code](https://github.com/liferay/liferay-portal/blob/c9d28ea3ee11a0d88000236d317e2d102e9290d9/portal-web/docroot/html/portal/render_portlet.jsp#L1038-L1071) is executing, implying that we want to refresh the page, but this too does not work.

In this PR we are directly refreshing the page on modal close, [here](https://github.com/liferay-frontend/liferay-portal/pull/31/commits/e6e83e2204242355024b96c35f61e06482052e4f#diff-21ae013e0a66817286d0bdb24786d327R263).

Is it worth looking to try to fix `portletAjaxable`?  What is our stance on it? Is it a legacy pattern or do we continue supporting it?

Reply:
Thanks @markocikos but I think @jbalsas meant that in [`e6e83e2204242355024b96c35f61e06482052e4f`](https://github.com/liferay-frontend/liferay-portal/pull/31/commits/e6e83e2204242355024b96c35f61e06482052e4f) the `<portlet:namespace />closeKaleoDialog` function was removed.

Now that the "workflow" modal/dialog is opened with [`openModal`](https://github.com/markocikos/liferay-portal/blob/301692e16e79a5016dd1d5fa7a8e3c6309f5e71c/modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/resources/META-INF/resources/admin/process/workflow.jsp#L260-L268)

I (locally) added the missing `<portlet:namespace />closeKaleoDialog` function, but I'm not able to close it with 

```javascript
Liferay.Util.getOpener().fire('closeModal');
```

I know if could / should probably add an `id` property to the call to `openModal` in  [`<portlet:namespace />editWorkflow`](https://github.com/markocikos/liferay-portal/blob/301692e16e79a5016dd1d5fa7a8e3c6309f5e71c/modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/resources/META-INF/resources/admin/process/workflow.jsp#L260-L268) to be able to pass it to that `closeModal` event, but I'm wonderning if I should revert back to [`35d9654e864f8bbae885bc298426a52d187070b0`](https://github.com/liferay-frontend/liferay-portal/pull/31/commits/35d9654e864f8bbae885bc298426a52d187070b0), until everything is clear. What do you think?

Reply:
> I think @jbalsas meant that in e6e83e2204242355024b96c35f61e06482052e4f the <portlet:namespace />closeKaleoDialog function was removed.

In `closeKaleoDialog` function the only thing that we doing is destroying the modal dialog instance. This no longer needed with `openModal`:
1. There is no `dialog` instance, there is no concept of a component instance.
1. We are always destroying and recreating modal content, this is an internal mechanism. There is no way to hide and show `openModal` dialog through API.

>I (locally) added the missing <portlet:namespace />closeKaleoDialog function, but I'm not able to close it with `closeModal`

We don't need to fire `closeModal` manually at any point, the three points of closing modal all all standard and work out-of-the-box:
1. (x) icon on top right of the modal.
1. A button with `type='cancel'`
1. Starting from this PR, an element with class `lfr-hide-dialog`, matching the link in success message.

Reply:
Thanks @markocikos, that does it for me! :+1: